### PR TITLE
Add automatic tests for RESTful API

### PR DESCRIPTION
**Pull request by Diman0**
and fix all remaining issues that I could find with the API.

## What type of PR?

internal feature / bug-fix

## What does this PR do?
I first wanted to finish #3113 before continuing on the tests to keep the scope smaller of the PR.

This PR adds automatic tests that tests **all** the interfaces of the RESTful API. Practically it only tests the normal Ok (http 200) situations. Maybe in the future we could add more tests to check if the validation checks work correctly for each interface.

I also fixed any issues I could find with the RESTful API. I can at least confirm that all interfaces work now. I think the validation checks are also complete now.

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [n/a] In case of feature or enhancement: documentation updated accordingly
- [n/a] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
bors try

**Comment by Diman0** [comment]
bors try

**Review by Diman0 — COMMENTED** [review]
I must change a data type in user.py

File: core/admin/mailu/api/v1/user.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
     'enable_pop': fields.Boolean(description='Allow email retrieval via POP3'),
     'allow_spoofing': fields.Boolean(description='Allow the user to spoof the sender (send email as anyone)'),
     'forward_enabled': fields.Boolean(description='Enable auto forwarding'),
-    'forward_destination': fields.List(fields.String(description='Email address to forward emails to'), example='Other@example.com'),
```
[the remark this answers is not in the document]
Comment (Diman0) [reply core/admin/mailu/api/v1/user.py:25]: I fixed all issues with the user form. I also added 2 missing checks in the api when creating/updating an user in regard to forward_destination.

**Comment by Diman0** [comment]
I fixed all issues with the user form. I also added 2 missing checks in the api when creating/updating an user in regard to forward_destination. The PR should now be ready for review.

bors try

**Review by nextgens — APPROVED** [review]
I really think we should move all these checks to the model... the API is undeniably better with these changes and we should get on with the release.

@Diman0 We need a changelog entry for this

**Pull request merged**